tests/python/unittest/test_random.py

```diff
@@ -552,6 +554,56 @@ def compute_expected_prob():
     mx.test_utils.assert_almost_equal(exp_cnt_sampled.asnumpy(), exp_cnt[sampled_classes].asnumpy(), rtol=1e-1, atol=1e-2)
     mx.test_utils.assert_almost_equal(exp_cnt_true.asnumpy(), exp_cnt[true_classes].asnumpy(), rtol=1e-1, atol=1e-2)
 
+@with_seed()
+def test_shuffle():
```

Comment:
Could you elaborate your reasoning of this unit test? How did you prove it's a uniformly random shuffling.

Reply:
It just count the probability of each possible outcomes. I added some comments explaining the test.